exchange/exchange.go

```diff
@@ -655,24 +646,25 @@ func getPrimaryAdServer(adServerId int) (string, error) {
 }
 
 // Extract all the data from the SeatBids and build the ExtBidResponse
-func (e *exchange) makeExtBidResponse(adapterBids map[openrtb_ext.BidderName]*pbsOrtbSeatBid, adapterExtra map[openrtb_ext.BidderName]*seatResponseExtra, req *openrtb.BidRequest, resolvedRequest json.RawMessage, errList []error) *openrtb_ext.ExtBidResponse {
+func (e *exchange) makeExtBidResponse(adapterBids map[openrtb_ext.BidderName]*pbsOrtbSeatBid, adapterExtra map[openrtb_ext.BidderName]*seatResponseExtra, req *openrtb.BidRequest, resolvedRequest json.RawMessage, debugInfo bool, errList []error) *openrtb_ext.ExtBidResponse {
 	bidResponseExt := &openrtb_ext.ExtBidResponse{
 		Errors:               make(map[openrtb_ext.BidderName][]openrtb_ext.ExtBidderError, len(adapterBids)),
 		ResponseTimeMillis:   make(map[openrtb_ext.BidderName]int, len(adapterBids)),
 		RequestTimeoutMillis: req.TMax,
 	}
-	if req.Test == 1 {
+	if debugInfo {
 		bidResponseExt.Debug = &openrtb_ext.ExtResponseDebug{
```

Comment:
I know this is not a change specific to this PR but I'd suggest we call `buildResolvedRequest` here itself if `debugInfo` is set rather than calling it upstream and passing it down because this is the only function that actually uses it.

This should be a simple and quick enough simplification of the code.

Reply:
I'd agree with you if `HoldAuction`'s second argument `bidRequest *openrtb.BidRequest` does not get modified anywhere from `exchange/exchange.go`'s line 100 all the way here. Let me double check because it'll be great if we could get rid of this extra argument in functions `makeExtBidResponse` and `buildBidResponse`